dataproc_jupyter_plugin/tests/test_dataproc.py

```diff
@@ -13,29 +13,54 @@
 # limitations under the License.
 
 import json
+from unittest import mock
 
 from dataproc_jupyter_plugin.tests import mocks
 
+from google.cloud.dataproc_v1.services.cluster_controller import (
+    ClusterControllerClient,
+    pagers
+)
+from google.auth import credentials as ga_credentials
+from google.cloud.dataproc_v1.types import clusters
+import pytest
 
-async def test_list_clusters(monkeypatch, jp_fetch):
-    mocks.patch_mocks(monkeypatch)
 
-    mock_project_id = "credentials-project"
-    mock_page_token = "mock-page-token"
-    mock_region_id = "mock-region"
-    mock_page_size = "mock_page_size"
-    response = await jp_fetch(
-        "dataproc-plugin",
-        "clusterList",
-        params={"pageSize": mock_page_size, "pageToken": mock_page_token},
-    )
-    assert response.code == 200
-    payload = json.loads(response.body)
-    assert (
-        payload["api_endpoint"]
-        == f"https://dataproc.googleapis.com//v1/projects/credentials-project/regions/{mock_region_id}/clusters?pageSize={mock_page_size}&pageToken={mock_page_token}"
+@pytest.mark.parametrize(
+    "request_type",
+    [
+        clusters.ListClustersRequest,
+        dict,
+    ],
+)
+
+def test_list_clusters(request_type, transport: str = "grpc"):
```

Comment:
This modified method does not actually test listing clusters via the server.

There needs to be an invocation of the `jp_fetch` method used to hit the `/dataproc-plugin/clusterList` endpoint, and the response from that call needs to be inspected to verify that it actually called into the underlying API client.